ee/control/client_http.go

```diff
@@ -0,0 +1,150 @@
+package control
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+)
+
+// HTTPClient handles retrieving control data via HTTP
+type HTTPClient struct {
+	logger     log.Logger
+	addr       string
+	baseURL    *url.URL
+	client     *http.Client
+	insecure   bool
+	disableTLS bool
+}
+
+// controlRequest is the payload sent in control server requests
+type controlRequest struct {
+	// TODO: This is a temporary and simple data format for phase 1
+	Message string `json:"message"`
+}
+
+func NewControlHTTPClient(addr string, opts ...HTTPClientOption) (*HTTPClient, error) {
+	baseURL, err := url.Parse(fmt.Sprintf("https://%s", addr))
+	if err != nil {
+		return nil, fmt.Errorf("parsing URL: %w", err)
+	}
+	c := &HTTPClient{
+		baseURL: baseURL,
+		client:  http.DefaultClient,
+		addr:    addr,
+	}
+
+	for _, opt := range opts {
+		opt(c)
+	}
+
+	return c, nil
+}
+
+func (c *HTTPClient) Get(subsystem, cachedETag string) (etag string, data io.Reader, err error) {
+	verb, path := "GET", "/api/v1/control"
+	params := &controlRequest{
+		Message: "ping",
+	}
+
+	response, err := c.do(verb, path, cachedETag, params)
+	if err != nil {
+		level.Error(c.logger).Log(
+			"msg", "error making request to control server endpoint",
+			"err", err,
+		)
+		return "", nil, err
+	}
+	defer response.Body.Close()
+
+	switch response.StatusCode {
+	case http.StatusNotFound:
+		// This could indicate an inconsistency in server data, or a client logic error
+		level.Error(c.logger).Log(
+			"msg", "got HTTP 404 making control server request",
+			"err", err,
+		)
+		return "", nil, err
```

Comment:
```suggestion
		)
		return "", nil, nil
```

`err` should be nil here.